File: src/main/java/com/unicorn/controller/MainController.java
package com.unicorn.controller;

import com.unicorn.service.FileService;
import com.unicorn.utils.Util;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import javax.annotation.Resource;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.PrintWriter;
import java.util.Properties;

/**
 * Created by dtimchenko on 02-Aug-17.
 */

@RestController
public class MainController {

    @Resource(name = "properties")
    private Properties properties;
    private FileService service;

    @Autowired
    public MainController(FileService fileService) {
        this.service = fileService;
    }

    @RequestMapping("/**")
    public void getResponse(HttpServletRequest request, HttpServletResponse response) {
        String requestURI = request.getRequestURI();
        String uri = properties.getProperty(requestURI);

        if (uri != null && !uri.isEmpty()) {
            String json = service.jsonFileContent(uri);
            response.setHeader("content-type", "application/json");
            PrintWriter writer = Util.getResponseWriter(response);
            if (writer != null) {
                writer.write(json);
                writer.close();
            }
        }
    }

//    private String readFile(String name) {
//        InputStream is = getClass().getResourceAsStream("./resources/"+name);
//        InputStreamReader reader = new InputStreamReader(is);
//        return new Scanner(reader).useDelimiter("\\Z").next();
//    }

}

File: src/main/java/com/unicorn/restemulator/controller/MainController.kt
package com.unicorn.restemulator.controller

import com.unicorn.restemulator.service.FileService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import java.io.PrintWriter
import java.util.*
import javax.annotation.Resource
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

/**
 * Created by dtimchenko on 10-Aug-17.
 */
const val EMPTY_JSON: String = "{}"


@RestController
class MainController @Autowired constructor(val fileService: FileService) {

    @Resource(name = "properties")
    lateinit var properties: Properties

    @RequestMapping(value = "/**")
    fun getResponse(request: HttpServletRequest, response: HttpServletResponse) {
        val requestURI = request.requestURI
        val jsonFile: String = properties.getProperty(requestURI) ?: ""

        val json = if (!jsonFile.isEmpty())
            fileService.getJsonFileContent(jsonFile)
        else
            EMPTY_JSON

        val writer: PrintWriter = response.writer
        response.setHeader("content-type", "application/json")
        writer.write(json)
        writer.close()
    }
}